xformers/components/multi_head_dispatch.py

```diff
@@ -143,11 +143,8 @@ def forward(
         self._check(value, "value")
         self._check(key, "key")
 
-        if query.size(0) < key.size(0):
-            query = query.expand_as(key)
-        elif query.size(0) > key.size(0):
-            key = key.expand_as(query)
-            value = value.expand_as(query)
+        max_batch = max((query.shape[0], key.shape[0], value.shape[0]))
+        query, key, value = map(lambda x: x.expand(max_batch, -1, -1), [query, key, value])
```

Comment:
As an opt, this could be (should be?) conditional on whether broadcasting is supported. WDYT?

Reply:
hmm I think it's fine like this, no ? almost a one liner, fixes some usecases without being an eye sore and it does not seem to break anything. I [checked the speed](https://app.circleci.com/pipelines/github/facebookresearch/xformers/914/workflows/6f29b158-d721-4deb-851a-8031a8f4b8d2/jobs/1823/parallel-runs/0/steps/0-111) and it does not break anything on this side either, so fine by me !

Maybe that I misunderstood your question though, but in general simple broadcasting should have worked, it broke here because when we resize QKV to cater for the heads and fold the heads into the batch dimension, we would not respect this batch difference (and your PR fixes that !). This popped up for a RL usecase, where the policy would generate a batch of trajectories but there was a single K to evaluate against (or related, cc @ludc )